spec.html

```diff
@@ -8488,11 +8509,10 @@ <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _this
       <p>The abstract operation OrdinaryFunctionCreate requires the arguments: an object _functionPrototype_, a parameter list Parse Node specified by _ParameterList_, a body Parse Node specified by _Body_, _thisMode_ which is either ~lexical-this~ or ~non-lexical-this~, and a Lexical Environment specified by _Scope_. OrdinaryFunctionCreate performs the following steps:</p>
       <emu-alg>
         1. Assert: Type(_functionPrototype_) is Object.
-        1. Let _F_ be a newly created ECMAScript function object with the internal slots listed in <emu-xref href="#table-27"></emu-xref>.
+        1. Let _internalSlotsList_ be the internal slots listed in <emu-xref href="#table-27"></emu-xref>.
+        1. Let _F_ be ! OrdinaryObjectCreate(_functionPrototype_, _internalSlotsList_).
         1. Set _F_'s essential internal methods to the default ordinary object definitions specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.
         1. Set _F_.[[Call]] to the definition specified in <emu-xref href="#sec-ecmascript-function-objects-call-thisargument-argumentslist"></emu-xref>.
```

Comment:
Given @jmdyck's `MakeConstructor` PRs, do we want to take the opportunity to spell out something more explicit about the verbiage around overriding the virtual internal methods? Internal slots are allocated ahead of time (which this patch makes very explicit), then may be "set" afterwards. Internal methods, OTOH, are currently "set" both to mean to override and to make present, e.g. [[Construct]] not being present until `MakeConstructor` installs it. This could use some tightening of the language.

Reply:
When you say "virtual internal methods", do you mean something other than just "internal methods"?

> Internal slots are allocated ahead of time (which this patch makes very explicit), then may be "set" afterwards.

This PR increases the 'distance' by one operation-call, but the explicitness is about the same.

(We could eliminate this separation by passing down not just an `_internalSlotsList_`, but (say) an `_internalSlotsRecord_`, with the name and initial value for each slot. Probably a separate PR though.)

Also, the dual-purpose meaning of "set" exists for spec entities other than internal methods, e.g. record-fields.